File: demo/Python/reverse_characters_only.py
def reverse_ignore_spaces(s: str) -> str:
    arr = list(s)
    left, right = 0, len(arr) -1 

    while left < right:
        if arr[left] == ' ':
            left +=1
        elif arr[right] == ' ':
            right-=1
        else:
            arr[left], arr[right], = arr[right], arr[left]
            left +=1
            right -=1

    return ''.join(arr)

if __name__ == "__main__":
    print(reverse_ignore_spaces("a b c"))

File: demo/Java/JavaSolutions/src/ReverseCharactersOnly.java
public class ReverseCharactersOnly{

    public static String reverseIgnoreSpace(String s){
        char [] arr = s.toCharArray();
        int left = 0, right = arr.length -1;

        while (left < right) {
            if (arr[left] == ' ') {
                left++;
            } else if (arr[right] == ' ') {
                right--;
            } else {
                char temp = arr[left];
                arr[left]  = arr[right];
                arr[right] = temp;
                left++;
                right--;
            }
        }
        return new String(arr);
    }

    public static void main(String[] args){
        System.out.println(reverseIgnoreSpace("a b c"));
    }
}